pendulum_controller/test/test_pendulum_controller_node.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Igor Recio
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+#include "pendulum_controller/pendulum_controller_node.hpp"
+
+using pendulum::pendulum_controller::PendulumControllerNode;
+
+using lifecycle_msgs::msg::State;
+using lifecycle_msgs::msg::Transition;
+
+class InitNodesTest : public ::testing::Test
+{
+protected:
+  std::vector<rclcpp::Parameter> params;
+  rclcpp::NodeOptions node_options;
+  std::vector<double> feedback_matrix = {-10.0000, -51.5393, 356.8637, 154.4146};
+
+  static void SetUpTestCase()
+  {
+    ASSERT_FALSE(rclcpp::ok());
+    rclcpp::init(0, nullptr);
+    ASSERT_TRUE(rclcpp::ok());
+  }
+
+  void SetUp() override
+  {
+    params.emplace_back("state_topic_name", "joint_states");
+    params.emplace_back("command_topic_name", "command");
+    params.emplace_back("teleop_topic_name", "setpoint");
+    params.emplace_back("command_publish_period_us", 10000);
+    params.emplace_back("enable_topic_stats", false);
+    params.emplace_back("topic_stats_topic_name", "controller_stats");
+    params.emplace_back("topic_stats_publish_period_ms", 1000);
+    params.emplace_back("deadline_duration_ms", 0);
+    params.emplace_back("controller.feedback_matrix", feedback_matrix);
+  }
+
+  static void TearDownTestCase()
+  {
+    (void)rclcpp::shutdown();
+  }
+};
+
+TEST_F(InitNodesTest, test_options_constructor) {
+  node_options.parameter_overrides(params);
+
+  const auto test_node =
```

Comment:
Just test the constructor does not throw with `ASSERT_NO_THROW` and test any possible bad configuration if there is any